Sources/GRPCInterceptors/Tracing/SpanAttributes+RPCAttributes.swift

```diff
@@ -0,0 +1,209 @@
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+internal import GRPCCore
+internal import Tracing
+
+@dynamicMemberLookup
+package struct RPCAttributes: SpanAttributeNamespace {
+  var attributes: SpanAttributes
+
+  init(attributes: SpanAttributes) {
+    self.attributes = attributes
+  }
+
+  struct NestedSpanAttributes: NestedSpanAttributesProtocol {
+    init() {}
+
+    var system: Key<String> { "rpc.system" }
+    var method: Key<String> { "rpc.method" }
+    var service: Key<String> { "rpc.service" }
+    var messageID: Key<Int> { "rpc.message.id" }
+    var messageType: Key<String> { "rpc.message.type" }
+    var grpcStatusCode: Key<Int> { "rpc.grpc.status_code" }
+
+    var serverAddress: Key<String> { "server.address" }
+    var serverPort: Key<Int> { "server.port" }
+
+    var clientAddress: Key<String> { "client.address" }
+    var clientPort: Key<Int> { "client.port" }
+
+    var networkTransport: Key<String> { "network.transport" }
+    var networkType: Key<String> { "network.type" }
+    var networkPeerAddress: Key<String> { "network.peer.address" }
+    var networkPeerPort: Key<Int> { "network.peer.port" }
+  }
+}
+
+extension SpanAttributes {
+  /// Semantic conventions for RPC spans.
+  package var rpc: RPCAttributes {
+    get {
+      .init(attributes: self)
+    }
+    set {
+      self = newValue.attributes
+    }
+  }
+}
+
+extension Span {
+  // See: https://opentelemetry.io/docs/specs/semconv/rpc/rpc-spans/
+  func setOTelClientSpanGRPCAttributes(
+    context: ClientContext,
+    serverHostname: String,
+    networkTransportMethod: String
+  ) {
+    self.attributes.rpc.system = "grpc"
+    self.attributes.rpc.serverAddress = serverHostname
+    self.attributes.rpc.networkTransport = networkTransportMethod
+    self.attributes.rpc.service = context.descriptor.service.fullyQualifiedService
+    self.attributes.rpc.method = context.descriptor.method
+
+    // Set server address information
+    switch PeerAddress(context.remotePeer) {
+    case .ipv4(let address, let port):
+      self.attributes.rpc.networkType = "ipv4"
+      self.attributes.rpc.networkPeerAddress = address
+      self.attributes.rpc.networkPeerPort = port
+      self.attributes.rpc.serverPort = port
+
+    case .ipv6(let address, let port):
+      self.attributes.rpc.networkType = "ipv6"
+      self.attributes.rpc.networkPeerAddress = address
+      self.attributes.rpc.networkPeerPort = port
+      self.attributes.rpc.serverPort = port
+
+    case .unixDomainSocket(let path):
+      self.attributes.rpc.networkType = "unix"
```

Comment:
Looking at the docs https://opentelemetry.io/docs/specs/semconv/attributes-registry/network/, it seems like network type should be either "ipv4"/"ipv6" and that "unix" should be a transport.

Reply:
Hm, so should we leave the network type empty when using UDS?

Reply:
Also unsure about setting the network transport here, since we're getting it as an argument when initialising the interceptor.

Reply:
>Also unsure about setting the network transport here, since we're getting it as an argument when initialising the interceptor.

What about this? Same argument as we don't want to override the server address; we don't want to override the network transport either.

Reply:
Sorry, forgot to reply to that one. I agree we shouldn't set it; let's leave that one up to the user.